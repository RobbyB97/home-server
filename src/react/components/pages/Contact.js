import React from 'react'
import {connect} from 'react-redux'



const Contact = ({dispatch, nav, title}) => (
	<div id="contact">
		Contact component
	</div>
)



const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav,
		title: state.title.title,
	}
}

export default connect(mapStateToProps)(Contact)