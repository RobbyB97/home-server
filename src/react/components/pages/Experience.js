import React from 'react'
import {connect} from 'react-redux'

import Job from '../content/Job'
import PrevPage from '../common/PrevPage'
import NextPage from '../common/NextPage'


export class Experience extends React.Component {
    
    render() {
        return (
            <div id="experience">
                
                Experience component
                <Job
                    title="Finalsite"
                    text="Job"
                    image="/dist/images/icons/finalsite.png">
                </Job>

                <PrevPage
                    text="Skills"
                    route="/skills/"
                    float={true}
                />

                <NextPage 
                    text="Projects"
                    route="/projects/"
                    float={true}
                />
            </div>  
        )
    }
}


const mapStateToProps = (state) => {
	return {
		nav: state.navigation.nav,
        title: state.title.title,
        splash: state.home.splash
	}
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Experience)