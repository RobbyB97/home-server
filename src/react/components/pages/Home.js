import React from 'react'
import {connect} from 'react-redux'

import Attribute from '../content/Attribute'


export class Home extends React.Component {

    render() {

        const fastAttribute = {
            image: "/dist/images/icons/speed_icon.png",
            title: "Fast",
            text: "Fully minified files and an optimized structure guarantee your website will load quickly, even on inconsistent internet connections."
        }

        const responsiveAttribute = {
            image: "/dist/images/icons/responsive_icon.png",
            title: "Responsive",
            text: "Mobile first development built up to 4k screens guarantees that your website will be elegant and functional on any device."
        }

        const beautifulAttribute = {
            image: "/dist/images/icons/paint_icon.png",
            title: "Beautiful",
            text: "A thorough, UX focused design process and open communication guarantees your website will look as beautiful as you envision it."
        }

        const accessibleAttribute = {
            image: "/dist/images/icons/accessibility_icon.png",
            title: "Accessible",
            text: "Full compliance with ADA standards guarantees that your website will be accessible and easy to use for everyone."
        }
        
        return (
            <div id="home">
                
                <div className={this.props.splash ? "home__background phone one": "home__background phone two"}></div>
                <div className={this.props.splash ? "home__background tablet one": "home__background tablet two"}></div>
                <div className={this.props.splash ? "home__background desktop one": "home__background desktop two"}></div>
                <div className={this.props.splash ? "home__background huge one": "home__background huge two"}></div>

                {!this.props.splash &&
                    <div className="home__attributes">
                        <Attribute {...fastAttribute} />
                        <Attribute {...responsiveAttribute} />
                        <Attribute {...beautifulAttribute} />
                        <Attribute {...accessibleAttribute} />
                    </div>
                }
                
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
      nav: state.navigation.nav,
      title: state.title.title,
      home: state.home.home,
      homePage: state.home.homePage,
      splash: state.home.splash
    }
}

const mapDispatchToProps = (dispatch) => ({

})


export default connect(mapStateToProps, mapDispatchToProps)(Home)