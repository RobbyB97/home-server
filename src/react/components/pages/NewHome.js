import React from 'react'


export class NewHome extends React.Component {
    render() {
        return (
            <div id="newHome">                
                <div className="newHome__content">
                    <div className="newHome__head">
                        <div className="newHome__name">
                            <h2>Robby Bergers</h2>
                        </div>
                        <div className="newHome__title">
                            <h3>Front End Developer</h3>
                        </div>
                    </div>

                    <div className="newHome__body">

                    </div>
                </div>

                <div className="newHome__backdrop"></div>
            </div>
        )
    }
}


export default NewHome