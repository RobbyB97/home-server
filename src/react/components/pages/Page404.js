import React from 'react'
import {NavLink} from 'react-router-dom'


export class Page404 extends React.Component {
    
    render() {
        return (
            <div id="page404">
                <div className="page404__backdrop"></div>
                <div className="page404__content">
                    <div className="page404__image">
                        <img src="/dist/images/wildcard.png" />
                    </div>

                    <div className="page404__text">
                        Uh oh, looks like this page doesn't exist.
                    </div>

                    <div className="page404__home-button">
                        <NavLink to="/">
                            <button>
                                Return home
                            </button>
                        </NavLink>
                    </div>
                </div>
            </div>
        )
    }
}


export default Page404