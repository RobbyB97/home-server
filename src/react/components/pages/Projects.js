import React from 'react'
import {connect} from 'react-redux'

import PrevPage from '../common/PrevPage'
import NextPage from '../common/NextPage'


export class Projects extends React.Component {

    render() {
        return (
            <div id="projects">
                Projects component
                
                <PrevPage
                    text="Experience"
                    route="/experience/"
                    float={true}
                />

                <NextPage 
                    text="Resume"
                    route="/resume/"
                    float={true}
                />
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
      nav: state.navigation.nav,
      title: state.title.title,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {

    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Projects)