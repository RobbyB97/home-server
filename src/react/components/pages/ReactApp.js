import React from 'react'

import Header from '../Header'
import StyleGuide from './StyleGuide'
import Card from '../Card'

import {Link, NavLink} from 'react-router-dom'

export default class ReactApp extends React.Component {

    render() {
        return (
            <div>
                <section className="space home-cover"></section>
                <div className="home-text one">
                        <h2>Home-text 1</h2>
                    </div>
                    <div className="home-text two">
                        <h2>Home-text 2</h2>
                    </div>
                    <Card
                        link="https://www.github.com/robbyb97"
                        image="/assets/images/favicon.png"
                        title="My Github"
                        subtext="View my projects"
                    />
                <div className="home-cover">
                </div>
            </div>
        )
    }

}