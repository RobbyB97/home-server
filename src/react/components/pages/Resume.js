import React from 'react'

import PrevPage from '../common/PrevPage'
import NextPage from '../common/NextPage'


export class Resume extends React.Component {
    render() {
        return (
            <div id="resume">
                <object data="/resume.pdf">
                    <p>Looks like your browser doesn't support PDF files.</p>
                    <p><a href="/resume.pdf" download>Download instead.</a></p>
                </object>

                <PrevPage
                    text="Projects"
                    route="/projects"
                    float={true}
                />

                <div className="nextpage float">
                    <a href="mailto:bergersr@my.easternct.edu">
                        <p className="nextpage__text">
                            Contact me â†’
                        </p>
                    </a>
                </div>
            </div>
        )
    }
}

export default Resume