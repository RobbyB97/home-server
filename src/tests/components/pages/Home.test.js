import React from 'react'
import {shallow} from 'enzyme'

import {Home} from '../../../react/components/pages/Home'


test('Render Home page', () => {
    const wrapper = shallow(<Home/>)

    expect(wrapper).toMatchSnapshot()
})