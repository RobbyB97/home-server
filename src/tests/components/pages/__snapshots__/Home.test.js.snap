// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home page 1`] = `
<div
  id="home"
>
  <div
    className="home__backdrop"
  />
  <div
    className="home__attributes"
  >
    <Attribute
      image="/dist/images/icons/speed_icon.png"
      text="Fully minified files and an optimized structure guarantee your website will always load quickly."
      title="Fast"
    />
    <Attribute
      image="/dist/images/icons/responsive_icon.png"
      text="Mobile first development ensures your site works smoothly on all screen sizes."
      title="Responsive"
    />
    <Attribute
      image="/dist/images/icons/paint_icon.png"
      text="A UX focused design process guarantees your website is as functional as it is elegant."
      title="Beautiful"
    />
    <Attribute
      image="/dist/images/icons/accessibility_icon.png"
      text="Compliance with ADA standards ensures your website is accessible and easy to use for everyone."
      title="Accessible"
    />
  </div>
  <Divider />
  <NextPage
    float={true}
    route="/skills/"
    text="Skills"
  />
</div>
`;
